rotkehlchen/data_migrations/manager.py

```diff
@@ -0,0 +1,50 @@
+import logging
+from typing import TYPE_CHECKING, Callable, Dict, Any, NamedTuple, Optional
+
+from rotkehlchen.data_migrations.upgrades.upgrade_1 import data_upgrade_1
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+
+if TYPE_CHECKING:
+    from rotkehlchen.rotkehlchen import Rotkehlchen
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class UpgradeRecord(NamedTuple):
+    version: int
+    function: Callable
+    kwargs: Optional[Dict[str, Any]] = None
+
+
+UPGRADES_LIST = [
+    UpgradeRecord(version=1, function=data_upgrade_1),
+]
+
+
+class RotkiUpgradeManager:
+
+    def __init__(self, rotki: 'Rotkehlchen'):
+        self.rotki = rotki
+
+    def maybe_migrate_data(self) -> None:
+        # get last upgrade setting
+        settings = self.rotki.data.db.get_settings()
+        current_upgrade = settings.last_data_migration
+        for upgrade in UPGRADES_LIST:
+            if current_upgrade < upgrade.version:
+                self._perform_upgrade(upgrade)
+                current_upgrade += 1
+        self.rotki.data.db.conn.cursor().execute(
+            'INSERT OR REPLACE INTO settings(name, value) VALUES(?, ?)',
+            ('last_data_migration', current_upgrade),
+        )
+        self.rotki.data.db.conn.commit()
+
+    def _perform_upgrade(self, upgrade: UpgradeRecord) -> None:
+        try:
+            kwargs = upgrade.kwargs if upgrade.kwargs is not None else {}
+            upgrade.function(rotki=self.rotki, **kwargs)
+        except BaseException as e:
```

Comment:
Perhaps add a lgtm ignore warning (the specific one, both here and for db upgrade manager)